### PR TITLE
Add PML Studio support

### PR DESCRIPTION
**Pull request by yang827**
I'm using AVEVA PML Studio 1.1 as the code editor, so I've added PML Studio support. 
setting
pmlstudio

### CODE REVIEW

File: PmlUnit/CodeEditor.cs [hunk 1]
```diff
@@ -83,6 +85,7 @@ enum CodeEditorKind
         Other = 0,
         Atom = 0x41544f4d, // ATOM
         NotepadPlusPlus = 0x4e502b2b, // NP++
+        PMLStudio = 0x504d4c53, // PMLS
```
Comment (PoByBolek) [inline PmlUnit/CodeEditor.cs]: rename to `PmlStudio`
```suggestion
        PmlStudio = 0x504d4c53, // PMLS
```

File: PmlUnit/CodeEditor.cs [hunk 2]
```diff
@@ -274,6 +277,22 @@ protected override IEnumerable<string> GetExtraArguments(string fileName, int li
         }
     }
 
+    class PMLStudioCodeEditor : BaseCodeEditor
```
Comment (PoByBolek) [inline PmlUnit/CodeEditor.cs]: rename to `PmlStudioCodeEditor`
```suggestion
    class PmlStudioCodeEditor : BaseCodeEditor
```

File: PmlUnit/CodeEditor.cs [hunk 3]
```diff
@@ -274,6 +277,22 @@ protected override IEnumerable<string> GetExtraArguments(string fileName, int li
...
+            : base(fileName, arguments)
+        {
+        }
+
+        protected override IEnumerable<string> GetExtraArguments(string fileName, int lineNumber)
+        {
+            if (lineNumber > 0)
+                return new string[] { fileName };
+            else
+                return new string[] { fileName };
```
Comment (PoByBolek) [inline PmlUnit/CodeEditor.cs]: remove condition if both branches do exactly the same
```suggestion
            return new string[] { fileName };
```

**Comment by PoByBolek** [comment]
Oh, right, I totally forgot about PML Studio. Thanks for this!

Do you have any links to some public documentation that we could put in the README? Or all they all hidden in AVEVA's support center?

**Comment by yang827** [comment]
ID is required to download from AVEVA's official website. I have downloaded it and it is no longer updated. PML Studio requires the Microsoft Visual Studio 2010 Shell (Isolated), which has the Microsoft download link.

Microsoft Visual Studio 2010 Shell (Isolated)
https://aka.ms/VS2010/IsoShell/en-US

AVEVA PML Studio 1.1
[User Guide.pdf](https://github.com/PoByBolek/PmlUnit/files/4743274/User.Guide.pdf)
[52369.zip](https://github.com/PoByBolek/PmlUnit/files/4743275/52369.zip)

**Comment by yang827** [comment]
Sorry, I pushed the local configuration.Only file PmlUnit/CodeEditor.cs needs to be merged.
I have fixed the PML Studio issue:
The Cursor to the specified line.
https://docs.microsoft.com/en-us/visualstudio/ide/reference/go-to-command?view=vs-2019

**Comment by PoByBolek** [comment]
merged in 811b8fa2ac9985d1406d8aece96911926a942b52

**Comment by PoByBolek** [comment]
Thanks again @yang827, I really appreciate your work!

Just to nit-pick a little bit: could you base your next pull request on [develop](https://github.com/PoByBolek/PmlUnit/tree/develop)? (This one was still based on [master](https://github.com/PoByBolek/PmlUnit/tree/master).)

**Comment by yang827** [comment]
Ok, it's a pleasure to be a part of this. I've learned a lot from you.

**Pull request closed without merging**